src/crypto/sha256.cpp

```diff
@@ -2,8 +2,6 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <config/bitcoin-config.h> // IWYU pragma: keep
```

Comment:
In commit ff368465a66298487a88d8da575f5251d094d687

Could you remind me what's going on here, please? I assume these aren't needed because we're meant to be adding compile definitions to specific libs. But from what I can tell, we're not actually adding them to `sha256.cpp`? So... is this currently using an un-optimized sha2?

Generally I'd feel more comfortable turning these into opt-outs, I think, to avoid accidentally building without optims.

For the sake of review/sanity, I think for this commit it's worth documenting (in the commit message) exactly which defines are affected and where in CMake they've been added.

Reply:
> But from what I can tell, we're not actually adding them to `sha256.cpp`? So... is this currently using an un-optimized sha2?

Please note the `PUBLIC` keyword, which propagates definition as a usage requirement:
https://github.com/bitcoin/bitcoin/blob/2b7c0f4fdb81ec1cf0579fa71b37f56672934ab5/src/crypto/CMakeLists.txt#L125

and

https://github.com/bitcoin/bitcoin/blob/2b7c0f4fdb81ec1cf0579fa71b37f56672934ab5/src/crypto/CMakeLists.txt#L128

where that usage requirement is actually applied.

> For the sake of review/sanity, I think for this commit it's worth documenting (in the commit message) exactly which defines are affected and where in CMake they've been added.

Sure thing! I'll update the commit message during the next branch update.

Reply:
Please see https://github.com/hebasto/bitcoin/pull/269.

Reply:
> Hmm https://corecheck.dev/bitcoin/bitcoin/pulls/30454 is down again

It is up now :)